cmd/snap/cmd_run.go

```diff
@@ -298,7 +298,7 @@ func migrateXauthority(info *snap.Info) (string, error) {
 
 	// If our target directory (XDG_RUNTIME_DIR) doesn't exist we
 	// don't attempt to create it.
-	baseTargetDir := filepath.Join(dirs.XdgRuntimeDirBase, u.Uid)
+	baseTargetDir := filepath.Join(dirs.XdgRuntimeDirBase, strconv.FormatUint(uint64(u.UID()), 10))
```

Comment:
Adding String() helpers to UserID and GroupID would make this and the other places with similar `strconv.Format..` is call a bit nicer

```
func (u UserID) String() string {
    return strconv.FormatUint(uint64(u), 10)
}

func (g GroupID) String() string {
    return strconv.FormatUint(uint64(g), 10)
}
```